lib/action_dispatch/session/active_record_store.rb

```diff
@@ -90,29 +96,52 @@ def set_session(env, sid, session_data, options)
             sid
           end
         end
+        alias_method :write_session, :set_session
 
-        def destroy_session(env, session_id, options)
+        def destroy_session(request, session_id, options)
           logger.silence_logger do
-            if sid = current_session_id(env)
-              get_session_model(env, sid).destroy
+            if sid = current_session_id(request)
+              get_session_model(request, sid).destroy
+              env = request.is_a?(ActionDispatch::Request) ? request.env : request
               env[SESSION_RECORD_KEY] = nil
             end
 
             generate_sid unless options[:drop]
           end
         end
+        alias_method :delete_session, :destroy_session
 
-        def get_session_model(env, sid)
-          if env[ENV_SESSION_OPTIONS_KEY][:id].nil?
-            env[SESSION_RECORD_KEY] = find_session(sid)
-          else
-            env[SESSION_RECORD_KEY] ||= find_session(sid)
+        def get_session_model(request, id)
+          logger.silence_logger do
+            model = @@session_class.find_by_session_id(id)
+            if !model
+              id = generate_sid unless ActiveRecord.version.to_s.start_with?('4.')
```

Comment:
Seems dangerous to compare this to the to_s in case that implementation changes. Not sure of best practice here but it might be better to go with `ActiveRecord::VERSION::MAJOR == 4`